tests/all/limits.rs

```diff
@@ -0,0 +1,286 @@
+use anyhow::Result;
+use std::cell::{Cell, RefCell};
+use std::rc::Rc;
+use wasmtime::*;
+
+#[test]
+fn test_limits() -> Result<()> {
```

Comment:
Could the tests here be expanded with tests where the initial memory/table size exceed the maximum? Additionally API requests like `Memory::new` and `Table::new` should fail as well if they exceed the maximum? (ideally also tests where API-created objects will also fail to `grow` past the maximum)